### ISSUE TITLE
[Bug]: [libil2cpp.so] il2cpp::vm::Method::GetParamCount(MethodInfo const*)

### ISSUE THREAD

**Issue by shefich** [OneSignal/OneSignal-Unity-SDK#607]
### What happened?

Crash happened don't know how to fix it. Can't reproduce on my end.

### Steps to reproduce?

```Markdown
1. install OneSignal 3.0.11 SDk for unity
```

### What did you expect to happen?

No to see the crashes in the Google Play.

### Unity version

2021.3.25

### OneSignal Unity SDK version

3.0.11

### Platform

Android

### Relevant log output

```Shell
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 3052 >>> com.xxx.xxx <<<

backtrace:
  #00  pc 0x0000000000571d9c  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/lib/arm64/libil2cpp.so (il2cpp::vm::Method::GetParamCount(MethodInfo const*))
  #01  pc 0x0000000000264200  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/lib/arm64/libunity.so (scripting_method_invoke(ScriptingMethodPtr, ScriptingObjectPtr, ScriptingArguments&, ScriptingExceptionPtr*, bool))
  #02  pc 0x0000000000271f0c  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/lib/arm64/libunity.so (ScriptingInvocation::Invoke(ScriptingExceptionPtr*, bool))
  #03  pc 0x00000000000ce7c8  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/lib/arm64/libunity.so (UnityJavaProxy_invoke(_JNIEnv*, _jobject*, long, _jstring*, _jobjectArray*))
  #04  pc 0x00000000000a91b0  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.odex (art_jni_trampoline+208)
  #05  pc 0x0000000000148570  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.odex (com.unity3d.player.ReflectionHelper.a+48)
  #06  pc 0x00000000001385b8  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_static_stub+568)
  #07  pc 0x00000000001470cc  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+276)
  #08  pc 0x00000000002eb4f8  /apex/com.android.runtime/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+384)
  #09  pc 0x00000000002e6584  /apex/com.android.runtime/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+900)
  #10  pc 0x00000000005ad918  /apex/com.android.runtime/lib64/libart.so (MterpInvokeStatic+368)
  #11  pc 0x0000000000132994  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_static+20)
  #12  pc 0x0000000000b9dea8  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.vdex (com.unity3d.player.ReflectionHelper$1.invoke+58)
  #13  pc 0x00000000002bbe28  /apex/com.android.runtime/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.5467380822148485331)+240)
  #14  pc 0x000000000059c210  /apex/com.android.runtime/lib64/libart.so (artQuickToInterpreterBridge+1024)
  #15  pc 0x0000000000141468  /apex/com.android.runtime/lib64/libart.so (art_quick_to_interpreter_bridge+88)
  #16  pc 0x00000000001c07ac  /system/framework/arm64/boot.oat (java.lang.reflect.Proxy.invoke+92)
  #17  pc 0x00000000001385b8  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_static_stub+568)
  #18  pc 0x00000000001470cc  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+276)
  #19  pc 0x00000000004bad28  /apex/com.android.runtime/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
  #20  pc 0x00000000004bb9e8  /apex/com.android.runtime/lib64/libart.so (art::InvokeWithJValues(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, jvalue const*)+400)
  #21  pc 0x00000000003b85f8  /apex/com.android.runtime/lib64/libart.so (art::JNI::CallStaticObjectMethodA(_JNIEnv*, _jclass*, _jmethodID*, jvalue const*)+616)
  #22  pc 0x000000000057bf1c  /apex/com.android.runtime/lib64/libart.so (art::InvokeProxyInvocationHandler(art::ScopedObjectAccessAlreadyRunnable&, char const*, _jobject*, _jobject*, std::__1::vector<jvalue, std::__1::allocator<jvalue>>&)+660)
  #23  pc 0x000000000059d228  /apex/com.android.runtime/lib64/libart.so (artQuickProxyInvokeHandler+1056)
  #24  pc 0x0000000000140efc  /apex/com.android.runtime/lib64/libart.so (art_quick_proxy_invoke_handler+76)
  #25  pc 0x0000000000138334  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_stub+548)
  #26  pc 0x00000000001470ac  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+244)
  #27  pc 0x00000000004bad28  /apex/com.android.runtime/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
  #28  pc 0x00000000004bc770  /apex/com.android.runtime/lib64/libart.so (art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+1472)
  #29  pc 0x0000000000447e90  /apex/com.android.runtime/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+48)
  #30  pc 0x00000000000c8d34  /system/framework/arm64/boot.oat (art_jni_trampoline+180)
  #31  pc 0x0000000000138334  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_stub+548)
  #32  pc 0x00000000001470ac  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+244)
  #33  pc 0x00000000002eb4f8  /apex/com.android.runtime/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+384)
  #34  pc 0x00000000002e6584  /apex/com.android.runtime/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+900)
  #35  pc 0x00000000005b0f9c  /apex/com.android.runtime/lib64/libart.so (MterpInvokeVirtualQuick+596)
  #36  pc 0x0000000000136594  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_virtual_quick+20)
  #37  pc 0x0000000000b8b3aa  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.vdex (com.onesignal.OSObservable$1.run+24)
  #38  pc 0x00000000005ac874  /apex/com.android.runtime/lib64/libart.so (MterpInvokeInterface+1740)
  #39  pc 0x0000000000132a14  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_interface+20)
  #40  pc 0x0000000000b7bcfa  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.vdex (com.onesignal.CallbackThreadManager$Companion$runOnPreferred$1.invoke+4)
  #41  pc 0x00000000005b12a0  /apex/com.android.runtime/lib64/libart.so (MterpInvokeVirtualQuick+1368)
  #42  pc 0x0000000000136594  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_virtual_quick+20)
  #43  pc 0x0000000000b7bcd4  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.vdex (com.onesignal.CallbackThreadManager$Companion$runOnPreferred$1.invoke)
  #44  pc 0x00000000005ac874  /apex/com.android.runtime/lib64/libart.so (MterpInvokeInterface+1740)
  #45  pc 0x0000000000132a14  /apex/com.android.runtime/lib64/libart.so (mterp_op_invoke_interface+20)
  #46  pc 0x0000000000bf1ee6  /data/app/com.xxx.xxx-vV2KD9wfiHUuz8103OysGw==/oat/arm64/base.vdex (kotlin.concurrent.ThreadsKt$thread$thread$1.run+4)
  #47  pc 0x00000000002bbe28  /apex/com.android.runtime/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.llvm.5467380822148485331)+240)
  #48  pc 0x000000000059c210  /apex/com.android.runtime/lib64/libart.so (artQuickToInterpreterBridge+1024)
  #49  pc 0x0000000000141468  /apex/com.android.runtime/lib64/libart.so (art_quick_to_interpreter_bridge+88)
  #50  pc 0x0000000000138334  /apex/com.android.runtime/lib64/libart.so (art_quick_invoke_stub+548)
  #51  pc 0x00000000001470ac  /apex/com.android.runtime/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+244)
  #52  pc 0x00000000004bad28  /apex/com.android.runtime/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
  #53  pc 0x00000000004bbdd8  /apex/com.android.runtime/lib64/libart.so (art::InvokeVirtualOrInterfaceWithJValues(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, jvalue const*)+416)
  #54  pc 0x00000000004fc9f0  /apex/com.android.runtime/lib64/libart.so (art::Thread::CreateCallback(void*)+1176)
  #55  pc 0x00000000000e3b24  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+36)
  #56  pc 0x0000000000085330  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by shepherd-l** [comment]
Has this started happening on the most recent version of the Unity SDK? How frequently is the crash occurring?

Could you try reproducing the issue and provide the steps when you are able to do so?

**Comment by shefich** [comment]
The same situation as in https://github.com/OneSignal/OneSignal-Unity-SDK/issues/606
I don't know what happened, as I see it is connected to the "PromptForPushNotificationPermissionResponseHandler" and probably somehow connected to the Unity 2021.3.25. I started to see these issues after upgrading from Unity 2020 to the Unity 2021.3.25.
Android versions: 7,8,9,10,11,12, 13.

**Comment by vfilho** [comment]
We are seeing the same issue after migrate from Firebase to OneSignal. We can't reproduce the issue internally as well.

**Comment by pawegio** [comment]
Any updates on the issue? We're also encountering it on `3.0.11` and Unity `2022.3.4f1`.

**Comment by fuatcoskun** [comment]
We face the same issue with Unity 2022.3.6f1 & OneSignal SDK 3.0.11.
Is it fixed with 5.0.x?

**Comment by shepherd-l** [comment]
Apologies that this issue has gone stale. I haven’t seen any further reports of this. Please try upgrading the OneSignal SDK if you or anyone is still having this issue. If this is still an issue, please open a new report with updated information.
